src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Tests to ensure that -h does not return error when run. Also ensures that help
+text exists. This only tests for commands/subgroups as specified by the user."""
+
+from __future__ import print_function
+import unittest
+from sys import stderr
+from subprocess import (Popen, PIPE)
+
+class HelpTextTests(unittest.TestCase):
+    """Tests which ensure that help texts do not error out and includes help text."""
+
+    def _validate_output_validate_line(self, #pylint: disable=too-many-arguments, inconsistent-return-statements
+                                       command_input, line,
+                                       section, subgroups, commands,
+                                       subgroups_index, commands_index):
+        """ Read a line of text and see if the line designates a new section in the help text.
+        Parameter line (string) should be unprocessed. For example, the line should not be
+        stripped of starting or trailing white spaces.
+
+        This method returns the updated values of subgroups_index and commands_index as a tuple.
+        Tuple has ordering (subgroups_index, commands_index).
+        If an error occurs during validation, an assert is called. """
+
+        line = line.strip()
+
+        if (section == 'Command') or (section == 'Group'):
+            # if the line starts with the inputted command, then it describes the command.
+            # make sure the line has text after it
+            if line.startswith(command_input):
+                self.assertGreater(
+                    len(line),
+                    len(command_input),
+                    msg='Validating help output failed on line: ' + line)
+
+            return (subgroups_index, commands_index)
+
+        elif section == 'Arguments':
+            # For lines that start with '--' (for argument descriptions), make sure that
+            # there is something after the argument declaration
+            if line.startswith('--'):
```

Comment:
I believe also it's possible to start with `-`, for arguments like `-t` which is timeout usually.

Reply:
Timeout has format: `--timeout -t            : Server timeout in seconds.  Default: 60.` 

However, changing it to check for `-` just in case.

Reply:
Actually, keeping the search for '--' because I've not seen arguments start with -, but some lines do because that's how they wrapped.

Reply:
Can they also start with `-` with wrapping?